nav2_regulated_pure_pursuit_controller/src/regulated_pure_pursuit_controller.cpp

```diff
@@ -447,13 +459,14 @@ bool RegulatedPurePursuitController::inCollision(const double & x, const double
     return false;
   }
 
-  unsigned char cost = costmap_->getCost(mx, my);
-
-  if (costmap_ros_->getLayeredCostmap()->isTrackingUnknown()) {
-    return cost >= INSCRIBED_INFLATED_OBSTACLE && cost != NO_INFORMATION;
-  } else {
-    return cost >= INSCRIBED_INFLATED_OBSTACLE;
+  double footprint_cost_ = collision_checker_->footprintCostAtPose(
+    x, y, theta, costmap_ros_->getRobotFootprint());
```

Comment:
I will need to look into it in the next round of reviewing before merging, but figured it would be good to express this concern now in case it has merit to change sooner: Check what frame of reference the footprint cost at pose expects. What you're giving it is world coordinates, make sure this function expects world coordinates